src/bigunit.ts

```diff
@@ -555,13 +634,26 @@ export class BigUnit {
     const fractionalPartBigInt = BigInt(paddedFractionalPart);
 
     // Combine the integer and fractional parts
-    const combinedValueBigInt =
-      integerPartBigInt * BigInt(10 ** precision) + fractionalPartBigInt;
+    let combinedValueBigInt =
+      integerPartBigInt * precisionFactor + fractionalPartBigInt;
+
+    // If the integer part is negative, convert the fractional part to a negative number otherwise it will incorrectly be positive
+    if (integerPart.includes("-")) {
+      combinedValueBigInt = combinedValueBigInt * BigInt(-1);
+    }
 
     // Return the BigUnit
     return new BigUnit(combinedValueBigInt, precision, name);
   }
 
+  /**
+   * @description Convert a decimal string to a BigUnit, eg. "1.23456" in an overflow-safe way. This is to avoid BigInt(10 ** precision) which can overflow.
+   * @param decimalStringValue
+   */
+  private static bigIntPowerOfTen(precision): bigint {
```

Comment:
Why don't we just convert the precision to a `bigint` first? Then no need for string manipulation which is risky and computationally expensive.